### PR TITLE
other-metadata: LICENSE

### PR DESCRIPTION
**Pull request by mgorny**
Closes: https://bugs.gentoo.org/709020
Signed-off-by: Michał Górny <mgorny@gentoo.org>

### CODE REVIEW

**Review by SoapGentoo — APPROVED** [review]

File: other-metadata.rst [hunk 1]
```diff
@@ -94,4 +94,55 @@ this circumstance, and they will not fail for users.
...
+:Reported: no
+
+The ``LICENSE`` variable must explicitly list licenses for all files
+installed by the package.  If some of the applicable licenses are
+conditional to USE flags, appropriate USE conditionals need to
+be expressed in the variable.
+
+If the package bundles any dependencies that are either installed,
+statically linked or in any other way combined with installed files,
+the licenses of those dependencies need to be listed as well.  This
```
Comment (ulm) [inline other-metadata.rst]:
```suggestion
the licenses of these dependencies need to be listed as well.  This
```

File: other-metadata.rst [hunk 2]
```diff
@@ -94,4 +94,55 @@ this circumstance, and they will not fail for users.
...
+is not strictly required when statically linking to dependencies
```
Comment (ulm) [inline other-metadata.rst]:
```suggestion
is not required when statically linking to dependencies
```

I think we don't want packages to list licenses of any libraries that they're not installing.

File: other-metadata.rst [hunk 3]
```diff
@@ -94,4 +94,55 @@ this circumstance, and they will not fail for users.
...
+installed by separate Gentoo packages.
+
+The licenses for files that are not installed but that are used at build
+time are not listed explicitly.  However, they must not impose any
+restrictions that would prevent some of the users from legally building
+the package.
```
Comment (ulm) [inline other-metadata.rst]: What do you mean by this? Do we have any build tools that don't allow to build some things?

File: other-metadata.rst [hunk 4]
```diff
@@ -94,4 +94,55 @@ this circumstance, and they will not fail for users.
...
+
+*Rationale*: the primary purpose of the license support in package
+manager is to provide the users with ability to decide on acceptable
+licenses for their installed systems (and binary packages).  In order
+for this to work effectively, the packages must provide correct
```
Comment (ulm) [inline other-metadata.rst]:
```suggestion
for this to work effectively, the packages must provide a correct
```

File: other-metadata.rst [hunk 5]
```diff
@@ -94,4 +94,55 @@ this circumstance, and they will not fail for users.
...
+licensed package links to a more restrictively licensed dependency).
+This is enforced for bundled dependencies but not for static linking
+to other packages, as in the latter case it is non-trivial to implement
+and the package manager already verifies the license while building
+dependencies (but not when installing binary packages).
+
+The ebuild format does not provide a separate variable to list licenses
+needed only at build time.  So far it has not been considered important
+enough to have one, as the relevant files exist only temporarily
+on user's system and they do not affect the runtime use of packages.
```
Comment (ulm) [inline other-metadata.rst]:
```suggestion
on the user's system and they do not affect the runtime use of packages.
```

File: other-metadata.rst (hunk 3 above)
Comment (mgorny) [reply other-metadata.rst]: I don't have a specific example. However, if license terms can prohibit you from using some program, then I suppose they can also prohibit you from running it at build time.

**Comment by mgorny** [comment]
Merged all the suggestions.

File: other-metadata.rst (hunk 3 above)
Comment (ulm) [reply other-metadata.rst]: IMHO this has little to do with LICENSE labelling. So I'd suggest to drop the sentence for now, especially if there isn't any concrete example.

(Also note that a license can only prevent me from running a program if I accept it first, but nothing forces me to accept a license unless it is an EULA. So we're talking about EULA build-time dependencies here.)

Comment (mgorny) [reply other-metadata.rst]: Removed it then.

File: other-metadata.rst [hunk 6]
```diff
@@ -94,4 +94,51 @@ this circumstance, and they will not fail for users.
...
+time are not listed explicitly.
+
+*Rationale*: the primary purpose of the license support in package
+manager is to provide the users with ability to decide on acceptable
+licenses for their installed systems (and binary packages).  In order
+for this to work effectively, the packages must provide a correct
+and complete license list.
+
+Static linking combines code from multiple packages, potentially covered
+by different licenses.  Listing all of them is the simplest way
```
Comment (robbat2) [inline other-metadata.rst]: `all of them` -> `listing all licenses`

File: other-metadata.rst [hunk 7]
```diff
@@ -94,4 +94,51 @@ this circumstance, and they will not fail for users.
...
+of ensuring that nothing is missed, as well as protecting against wrong
+derivative work licenses stated upstream (i.e. when a less restrictively
+licensed package links to a more restrictively licensed dependency).
+This is enforced for bundled dependencies but not for static linking
```
Comment (robbat2) [inline other-metadata.rst]: Start a new paragraph here. `\n\nListing of licenses is enforced for bundled dependencies...`

File: other-metadata.rst [hunk 8]
```diff
@@ -94,4 +94,51 @@ this circumstance, and they will not fail for users.
...
+
+The ``LICENSE`` variable must explicitly list licenses for all files
+installed by the package.  If some of the applicable licenses are
+conditional to USE flags, appropriate USE conditionals need to
+be expressed in the variable.
+
+If the package bundles any dependencies that are either installed,
+statically linked or in any other way combined with installed files,
+the licenses of these dependencies need to be listed as well.  This
+is not required when statically linking to dependencies
```
Comment (robbat2) [inline other-metadata.rst]: `This is not presently required`

**Review by robbat2 — APPROVED** [review]

**Comment by mgorny** [comment]
Thanks, Robin, I've applied all your suggestions.

File: other-metadata.rst [hunk 9]
```diff
@@ -94,4 +94,53 @@ this circumstance, and they will not fail for users.
...
+-------
+:Source: QA
+:Reported: no
+
+The ``LICENSE`` variable must explicitly list licenses for all files
+installed by the package.  If some of the applicable licenses are
+conditional to USE flags, appropriate USE conditionals need to
+be expressed in the variable.
+
+If the package bundles any dependencies that are either installed,
```
Comment (Zlogene) [inline other-metadata.rst]:
```suggestion
If a package bundles any dependencies that are either installed,
```

File: other-metadata.rst [hunk 10]
```diff
@@ -94,4 +94,53 @@ this circumstance, and they will not fail for users.
...
+statically linked or in any other way combined with installed files,
+the licenses of these dependencies need to be listed as well.  This
+is not presently required when statically linking to dependencies
+installed by separate Gentoo packages.
```
Comment (Zlogene) [inline other-metadata.rst]:
```suggestion
installed by separate packages in the repository.
```

File: other-metadata.rst [hunk 11]
```diff
@@ -94,4 +94,53 @@ this circumstance, and they will not fail for users.
...
+
+The licenses for files that are not installed but that are used at build
+time are not listed explicitly.
+
+*Rationale*: the primary purpose of the license support in package
```
Comment (Zlogene) [inline other-metadata.rst]:
```suggestion
*Rationale*: the primary purpose of the license support in the package
```

File: other-metadata.rst [hunk 12]
```diff
@@ -94,4 +94,53 @@ this circumstance, and they will not fail for users.
...
+Listing of licenses is enforced for bundled dependencies but not for
+static linking to other packages, as in the latter case it is
+non-trivial to implement and the package manager already verifies
+the license while building dependencies (but not when installing binary
+packages).
+
+The ebuild format does not provide a separate variable to list licenses
+needed only at build time.  So far it has not been considered important
+enough to have one, as the relevant files exist only temporarily
+on the user's system and they do not affect the runtime use of packages.
```
Comment (Zlogene) [inline other-metadata.rst]:
```suggestion
on the user's system and do not affect the runtime use of packages.
```

**Review by Zlogene — CHANGES_REQUESTED** [review]

**Comment by mgorny** [comment]
All applied.

**Review by Zlogene — APPROVED** [review]

**Review by SoapGentoo — APPROVED** [review]

**Review by ulm — APPROVED** [review]

**Review by akhuettel — APPROVED** [review]

**Pull request closed without merging**